### PR TITLE
fix(components/indicators): fix content roles

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2623253](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2623253)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1513/storybook/)

Component Storybooks:

- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/colorpicker-storybook/)
- [cypress-screenshots-datetime-storybook-e2e](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/cypress-screenshots-datetime-storybook-e2e/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/split-view-storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/1513/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1513/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1513/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1513/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1513/playground/)

**Review by Blackbaud-CoreyArcher — DISMISSED** [review]

File: libs/components/indicators/src/lib/modules/tokens/tokens.component.scss [hunk 1]
```diff
@@ -6,7 +6,7 @@ $sky-tokens-gutter-size: 2px;
   display: flex;
   flex-wrap: wrap;
   align-items: baseline;
-  margin: 0 -$sky-tokens-gutter-size;
+  margin: -1 * $sky-tokens-gutter-size;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/indicators/src/lib/modules/tokens/tokens.component.scss:9]: Should this be `0 (-1 * $sky-tokens-gutter-size)`?

File: libs/components/indicators/src/lib/modules/tokens/tokens.component.scss [hunk 2]
```diff
@@ -19,10 +19,14 @@ $sky-tokens-gutter-size: 2px;
     flex-grow: 1;
     flex-basis: 0px;
 
-    // Add a zero-width space to allow form controls to align
-    // properly to text baseline.
-    &:before {
-      content: "#{'\200B'}";
+    > div {
+      display: contents;
```
Comment (Blackbaud-SteveBrush) [inline libs/components/indicators/src/lib/modules/tokens/tokens.component.scss:23]: I'm seeing some warnings around accessibility with the `display: contents` feature. Should we be concerned about that? https://caniuse.com/?search=display%3A%20contents

File: libs/components/lookup/src/lib/modules/autocomplete/autocomplete.component.html [hunk 3]
```diff
@@ -32,47 +32,53 @@
 
 <ng-template #resultControlsTemplateRef>
   <div
-    *ngIf="searchText && (!showActionsArea || searchResults.length > 0)"
     class="sky-autocomplete-results"
+    [class.sky-screen-reader-only]="
```
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/src/lib/modules/autocomplete/autocomplete.component.html]: I might be misunderstanding the issue, but instead of refactoring, could we check if the autocomplete results are "open" like we're doing for the `aria-owns` property? https://github.com/blackbaud/skyux/blob/200b77425edc88388f5525428d59eeb0289baf0e/libs/components/lookup/src/lib/modules/autocomplete/autocomplete.component.html#L8

File: libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts [hunk 4]
```diff
@@ -3744,11 +3744,26 @@ describe('Lookup component', function () {
 
       it('should be accessible', async () => {
         fixture.componentInstance.ariaLabelledBy = 'my-lookup-label';
+        fixture.componentInstance.setSingleSelect();
```
Comment (Blackbaud-SteveBrush) [inline libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts:3747]: Since these changes involve the lookup library, I would put them in a separate pull request so that they're mentioned in the changelog.

File: libs/components/indicators/src/lib/modules/tokens/tokens.component.scss (hunk 1 above)
Comment (johnhwhite) [reply libs/components/indicators/src/lib/modules/tokens/tokens.component.scss:9]: I don't know if that was the intent, but the design we've been using has been `(0 - $sky-tokens-gutter-size)` and we see warnings about the ambiguity. I tried `0 -1 * $sky-tokens-gutter-size` but that added vertical space; the current rule computes to `margin: -2px`.

We have several places where we're negating a variable and those will break in a future version.

I can remove this change as it's not really related to the issue.

File: libs/components/indicators/src/lib/modules/tokens/tokens.component.scss (hunk 2 above)
Comment (johnhwhite) [reply libs/components/indicators/src/lib/modules/tokens/tokens.component.scss:23]: The Safari issues are concerning, the fix landed in Safari Technology Preview 174, but there is potential to break accessibility for prior versions. I don't have a good assessment of how it compares the current issue and would have to dig into that.

Chrome and Firefox implementations seem to be fine.

File: libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts (hunk 4 above)
Comment (johnhwhite) [reply libs/components/lookup/src/lib/modules/lookup/lookup.component.spec.ts:3747]: Removed the autocomplete changes.

File: libs/components/lookup/src/lib/modules/autocomplete/autocomplete.component.html (hunk 3 above)
Comment (johnhwhite) [reply libs/components/lookup/src/lib/modules/autocomplete/autocomplete.component.html]: Removed the autocomplete changes.

File: libs/components/indicators/src/lib/modules/tokens/tokens.component.scss (hunk 2 above)
Comment (johnhwhite) [reply libs/components/indicators/src/lib/modules/tokens/tokens.component.scss:23]: I've confirmed that the current implementation makes that area inaccessible in Safari.

File: libs/components/indicators/src/lib/modules/tokens/tokens.component.scss (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/indicators/src/lib/modules/tokens/tokens.component.scss:9]: Strange; I'm guessing the original intent for that statement was `0 -2px`, but it looks like it evaluates to `-2px` (like you mentioned). 🤷 We'll keep your change for now.

**Review by Blackbaud-SteveBrush — APPROVED** [review]
